File: client/Assets/Scripts/GamePlay/Launch.cs
using UnityEngine;
using System.Collections;

/// <summary>
/// 游戏启动
/// </summary>
public class Launch : MonoBehaviour {

    public static Launch inst = null;

    // build in 
    void Awake() {
        Debug.Log(" Launch Awake Begin......");
        // 保证只有一个， 重复创建的对象直接删掉
        if (null != inst) {
            GameObject.DestroyImmediate(gameObject);
            Debug.Log(" Launch Already Awaked !!! ");
            return;
        }

        inst = this;

        CoDelegator.Instance = gameObject.AddMissingComponent<CoDelegator>();
        DontDestroyOnLoad(gameObject);
    }

	// Use this for initialization
	void Start () {
        StartInit();
	}

    public void StartInit(){
        Game.Instance.Init();
    }

    //预留接口
    public void DestoryInit(){
        Game.Instance.Clear();
    }
}

File: Mahjong/assets/Script/Engine/Launch.js
cc.Class({
    extends: cc.Component,

    properties: {
        
    },

    // use this for initialization
    onLoad: function () {
        this.init();
        
        this.uiMgr.openWindow("LoginWindow");
    },
    
    init: function(){
        cc.log("start launch init");
        cc.Game = this;
        this.uiMgr = require('UIMgr')();
        this.uiMgr.initUI();
    },
});
